src/connectors/autocomplete/connectAutocomplete.ts

```diff
@@ -0,0 +1,171 @@
+import {
+  AlgoliaSearchHelper as Helper,
+  SearchResults,
+} from 'algoliasearch-helper';
+import escapeHits, { TAG_PLACEHOLDER } from '../../lib/escape-highlight';
+import {
+  checkRendering,
+  createDocumentationMessageGenerator,
+  noop,
+  warning,
+} from '../../lib/utils';
+import {
+  RendererOptions,
+  Renderer,
+  WidgetFactory,
+  Hits,
+  InstantSearch,
+  Unmounter,
+} from '../../types';
+
+const withUsage = createDocumentationMessageGenerator({
+  name: 'autocomplete',
+  connector: true,
+});
+
+interface AutocompleteIndex {
+  index: string;
+  hits: Hits;
+  results: SearchResults | undefined;
+}
+
+interface AutocompleteConnectorParams {
+  /**
+   * Escapes HTML entities from hits string values.
+   *
+   * @default `true`
+   */
+  escapeHTML?: boolean;
+}
+
+type RefineFunction = (query: string) => Helper;
+
+export interface AutocompleteRendererOptions<TAutocompleteWidgetParams>
+  extends RendererOptions<TAutocompleteWidgetParams> {
+  currentRefinement: string;
+  indices: AutocompleteIndex[];
+  instantSearchInstance: InstantSearch;
+  refine: RefineFunction;
+}
+
+export type AutocompleteRenderer<TAutocompleteWidgetParams> = Renderer<
+  AutocompleteRendererOptions<
+    AutocompleteConnectorParams & TAutocompleteWidgetParams
+  >
+>;
+
+export type AutocompleteWidgetFactory<
+  TAutocompleteWidgetParams
+> = WidgetFactory<AutocompleteConnectorParams & TAutocompleteWidgetParams>;
+
+export type AutocompleteConnector = <TAutocompleteWidgetParams>(
+  render: AutocompleteRenderer<TAutocompleteWidgetParams>,
+  unmount?: Unmounter
+) => AutocompleteWidgetFactory<TAutocompleteWidgetParams>;
+
+const connectAutocomplete: AutocompleteConnector = (
+  renderFn,
+  unmountFn = noop
+) => {
+  checkRendering(renderFn, withUsage());
+
+  return widgetParams => {
+    const { escapeHTML = true } = widgetParams || {};
+
+    warning(
+      !(widgetParams as any).indices,
+      `Since InstantSearch.js 4, \`connectAutocomplete\` infers the indices from the tree of widgets.
+The \`indices\` option is ignored.`
```

Comment:
We could print the alternative usage WDYT?

Reply:
WDYM? A link to the doc?

Reply:
Convert the usage of `indices` into `index` This what we did for [the `geoSearch` warnings](https://github.com/algolia/instantsearch.js/blob/509513c0feafaad522f6f18d87a441559f4aa050/src/connectors/geo-search/connectGeoSearch.js#L117-L132).